src/piton/labelers/omop_inpatient_admissions.py

```diff
@@ -0,0 +1,161 @@
+"""Labeling functions for OMOP data based on lab values."""
+from __future__ import annotations
+
+import datetime
+from typing import List
+
+from .. import Event, Patient
+from ..extension import datasets as extension_datasets
+from .core import (
+    Label,
+    Labeler,
+    LabelType,
+    TimeHorizon,
+    TimeHorizonEventLabeler,
+)
+from .omop import (
+    WithinVisitLabeler,
+    _get_all_children,
+    get_death_concepts,
+    get_inpatient_admission_concepts,
+    get_inpatient_admission_discharge_times,
+    get_inpatient_admission_events,
+)
+
+
+class DummyAdmissionDischargeLabeler(Labeler):
+    """Generate a placeholder Label at every admission and discharge time for this patient."""
+
+    def __init__(
+        self,
+        ontology: extension_datasets.Ontology,
+    ):
+        self.ontology = ontology
+
+    def label(self, patient: Patient) -> List[Label]:
+        labels: List[Label] = []
+        for (admission_time, discharge_time) in zip(
+            *get_inpatient_admission_discharge_times(self.ontology, patient)
+        ):
+            labels.append(Label(time=admission_time, value=True))
+            labels.append(Label(time=discharge_time, value=True))
+        return labels
+
+    def get_labeler_type(self) -> LabelType:
+        return "boolean"
+
+
+class InpatientReadmissionLabeler(TimeHorizonEventLabeler):
+    """
+    This labeler is designed to predict whether a patient will be readmitted within `time_horizon`
+    It explicitly does not try to deal with categorizing admissions as "unexpected" or not and is thus
+    not comparable to other work.
+
+    Prediction time: At discharge from an inpatient admission.
+    Time horizon: Interval of time after discharg of length `time_horizon`
+    Label: TRUE if patient has an inpatient admission within `time_horizon`
+
+    Defaults to 30-day readmission labeler,
+        i.e. `time_horizon = TimeHorizon(1 second, 30 days)`
+    """
+
+    def __init__(
+        self,
+        ontology: extension_datasets.Ontology,
+        time_horizon: TimeHorizon = TimeHorizon(datetime.timedelta(seconds=1), datetime.timedelta(days=30)),  # type: ignore
+    ):
+        self.ontology: extension_datasets.Ontology = ontology
+        self.time_horizon: TimeHorizon = time_horizon
+
+    def get_outcome_times(self, patient: Patient) -> List[datetime.datetime]:
+        """Return the start times of inpatient admissions."""
+        admission_events: List[Event] = get_inpatient_admission_events(
```

Comment:
Here, don't we need to skip the first admission_events? The outcome to this task can also be seen after the first visit, no?